offixsdk/src/main/java/org/aerogear/offixsdk/Extension.kt

```diff
@@ -0,0 +1,48 @@
+package org.aerogear.offixsdk
+
+import android.util.Log
+import com.apollographql.apollo.ApolloCall
+import com.apollographql.apollo.api.Operation
+import org.aerogear.offixsdk.persistence.Mutation
+import org.json.JSONObject
+
+val application = Application
+
+/*  Extension function on ApolloCall which will be used by the user while making a call request.
+
+   @param mutation whose data will be stored in the database if network connection is not there.
+   @param callback Callback which will handle the response or a failure exception.
+ */
+
+fun ApolloCall<Any>.offQueue(
+    mutation: com.apollographql.apollo.api.Mutation<Operation.Data, Operation.Data, Operation.Variables>,
+    callback: ApolloCall.Callback<Any>
+) {
+
+    if (isNetwork(getContext())) {
+
+        Log.e("Extension", " Network is there.")
+        this.enqueue(callback)
+
+    } else {
+
+        Log.e("Extension", " Network not there.")
+
+        /* If the user is offline:
+           1. Store the object in database.
+           2. Schedule a work manager to process that request in future. (This is scheduled from LoggingInterceptor class)
+        */
+
+        val operationID = this.operation().operationId()
```

Comment:
Execute operation once

Reply:
What exactly do you mean by this?

Reply:
operation() is executed 8 times in the row in code below

Reply:
@Lavanyagaur22 you can probably use `with(this.operation())` here